packages/pillarbox-playlist/test/pillarbox-playlist.spec.js

```diff
@@ -0,0 +1,350 @@
+      expect(posterSpy).toHaveBeenLastCalledWith(playlist[1].poster);
+    });
+
+    it('should not play next element on ended if autoadvance is disabled', async() => {
+      // Given
+      const srcSpy = vi.spyOn(player, 'src').mockImplementation(() => {});
+      const posterSpy = vi.spyOn(player, 'poster').mockImplementation(() => {});
+
+      // When
+      pillarboxPlaylist.load(...playlist);
+      pillarboxPlaylist.handleEnded();
+
+      // Then
+      expect(pillarboxPlaylist.hasPrevious()).toBeFalsy();
+      expect(pillarboxPlaylist.hasNext()).toBeTruthy();
+      expect(pillarboxPlaylist.items.length).toBe(4);
+      expect(pillarboxPlaylist.index).toBe(0);
+      expect(pillarboxPlaylist.currentItem).toBe(playlist[0]);
+      expect(srcSpy).toHaveBeenLastCalledWith(playlist[0].sources);
+      expect(posterSpy).toHaveBeenLastCalledWith(playlist[0].poster);
+    });
+  });
+
+  describe('shuffle', () => {
+    it('should randomize the order of playlist items', () => {
+      // Given
+      vi.spyOn(player, 'src').mockImplementation(() => {});
+      vi.spyOn(player, 'poster').mockImplementation(() => {});
+
+      // When
+      pillarboxPlaylist.load(...playlist);
+      pillarboxPlaylist.shuffle();
+
+      // Then
+      expect(playlist).not.toEqual(pillarboxPlaylist.item);
+      expect(playlist.length).toBe(pillarboxPlaylist.items.length);
+    });
+  });
+
+  describe('push', () => {
+    it('should push new items at the end of the playlist', async() => {
+      // Given
+      vi.spyOn(player, 'src').mockImplementation(() => {});
+      vi.spyOn(player, 'poster').mockImplementation(() => {});
+      const items = [
+        { sources: [{ src: 'fifth-source', type: 'test' }], poster: 'fifth-poster' },
+        { sources: [{ src: 'sixth-source', type: 'test' }], poster: 'sixth-poster' }
+      ];
+
+      // When
+      pillarboxPlaylist.load(...playlist);
+      pillarboxPlaylist.push(...items);
+
+      // Then
+      expect(pillarboxPlaylist.items.length).toBe(6);
+      expect(pillarboxPlaylist.index).toBe(0);
+      expect(pillarboxPlaylist.currentItem).toBe(playlist[0]);
+    });
+  });
+
+  describe('splice', () => {
+    it('should push new items at any point of the playlist', async() => {
+      // Given
+      vi.spyOn(player, 'src').mockImplementation(() => {});
+      vi.spyOn(player, 'poster').mockImplementation(() => {});
+
+      const items = [
+        { sources: [{ src: 'fifth-source', type: 'test' }], poster: 'fifth-poster' },
+        { sources: [{ src: 'sixth-source', type: 'test' }], poster: 'sixth-poster' }
+      ];
+
+      // When
+      pillarboxPlaylist.load(...playlist);
+      pillarboxPlaylist.splice(0, 0, ...items);
+
+      // Then
+      expect(pillarboxPlaylist.items.length).toBe(6);
+      expect(pillarboxPlaylist.index).toBe(2);
+      expect(pillarboxPlaylist.currentItem).toBe(playlist[0]);
+    });
+
+    it('should delete items at any point fo the playlist', async() => {
```

Comment:
do you mean `foo` instead of `fo` ?

Reply:
Close enough: `any point of the playlist`. Nice catch.